http.go

```diff
@@ -5,11 +5,52 @@ import (
 	"fmt"
 	"net"
 	"net/http"
-	"net/url"
 	"runtime/debug"
 	"strings"
 )
 
+const (
+	redaction = "********"
+)
+
+// Query fields whose value will redacted. Used by NewHttp.
+var QuerySecretFields = []string{"password", "passphrase", "passwd", "secret"}
+
+// Header fields whose value will redacted. Used by NewHttp.
+var HeaderSecretFields = []string{"Authorization"}
+
+func redactQuery(r *http.Request) string {
+	query := r.URL.Query()
+
+	for _, keyword := range QuerySecretFields {
+		for field := range query {
+			if field == keyword {
+				query[field] = []string{redaction}
```

Comment:
I think it'd be worth noting here that this does **not** mutate the `Values` object. Mostly because this wasn't obvious to me without looking at the source code of Go to determine that we weren't mutating anything. :)